### PR TITLE
Parse command line arguments prior to executing on Docker

### PR DESCRIPTION
**Pull request by Benzbeeb**

### CODE REVIEW

File: chain/byteexec/byteexec.go [hunk 1]
```diff
@@ -33,21 +35,33 @@ func writeFile(executable []byte) (string, string, error) {
...
 
 	return exec.CommandContext(ctx, filename, args...).Output()
 }
 
 // RunOnDocker runs the given executable in a new docker container.
-func RunOnDocker(executable []byte, timeOut time.Duration, args ...string) ([]byte, error) {
+func RunOnDocker(executable []byte, timeOut time.Duration, arg string) ([]byte, error) {
+	args, err := shellwords.Parse(arg)
+	if err != nil {
+		return []byte{0}, err
```
Comment (taobun) [inline chain/byteexec/byteexec.go]: return []byte{}

File: chain/byteexec/byteexec.go [hunk 2]
```diff
@@ -33,21 +35,33 @@ func writeFile(executable []byte) (string, string, error) {
 }
 
 // RunOnLocal spawns a new subprocess and runs the given executable. NOT SAFE!
-func RunOnLocal(executable []byte, timeOut time.Duration, args ...string) ([]byte, error) {
+func RunOnLocal(executable []byte, timeOut time.Duration, arg string) ([]byte, error) {
+	args, err := shellwords.Parse(arg)
+	if err != nil {
+		return []byte{0}, err
```
Comment (taobun) [inline chain/byteexec/byteexec.go]: return []byte{}

**Review by taobun — CHANGES_REQUESTED** [review]

File: chain/byteexec/byteexec.go [hunk 3]
```diff
@@ -33,21 +35,33 @@ func writeFile(executable []byte) (string, string, error) {
 }
 
 // RunOnLocal spawns a new subprocess and runs the given executable. NOT SAFE!
-func RunOnLocal(executable []byte, timeOut time.Duration, args ...string) ([]byte, error) {
+func RunOnLocal(executable []byte, timeOut time.Duration, arg string) ([]byte, error) {
+	args, err := shellwords.Parse(arg)
+	if err != nil {
+		return []byte{}, err
```
Comment (sorawit) [inline chain/byteexec/byteexec.go]: @taobun. I think return `nil` makes more sense. Thoughts?

File: chain/byteexec/byteexec_test.go [hunk 4]
```diff
@@ -13,5 +13,23 @@ func TestExecByte(t *testing.T) {
...
+	execByte, _ := hex.DecodeString("23212f62696e2f626173680a0a6563686f2024320a")
+	result, err := RunOnLocal(execByte, 10*time.Second, "aS world")
+	require.Nil(t, err)
+	require.Equal(t, []byte("world\n"), result)
+}
+
+// 23212f62696e2f626173680a0a6563686f2024322b24310a
+func TestExecByte3(t *testing.T) {
+	// batch script that can sum two number
+	execByte, _ := hex.DecodeString("23212f62696e2f626173680a6e756d313d24310a6e756d323d24320a6563686f202428286e756d312b6e756d3229290a")
```
Comment (sorawit) [inline chain/byteexec/byteexec_test.go:48]: Let's also have the script here as comment so tests are easier to follow.

**Review by sorawit — CHANGES_REQUESTED** [review]

File: chain/byteexec/byteexec.go (hunk 3 above)
Comment (taobun) [reply chain/byteexec/byteexec.go]: In my opinion, I prefer `nil`. But other codes in our repo use `[]byte{}`. How we will fix it?

Comment (sorawit) [reply chain/byteexec/byteexec.go]: Ok. let's use `nil` here then. We will deal with other places later.

**Review by sorawit — DISMISSED** [review]

File: chain/byteexec/byteexec.go (hunk 3 above)
Comment (taobun) [inline chain/byteexec/byteexec.go]: nil?

File: chain/byteexec/byteexec_test.go [hunk 5]
```diff
@@ -13,5 +13,28 @@ func TestExecByte(t *testing.T) {
 	execByte, _ := hex.DecodeString("23212f62696e2f626173680a0a6563686f2024310a")
 	result, err := RunOnLocal(execByte, 10*time.Second, "hello world")
 	require.Nil(t, err)
-	require.Equal(t, []byte("hello world\n"), result)
+	require.Equal(t, []byte("hello\n"), result)
+}
+
+func TestExecByte2(t *testing.T) {
```
Comment (taobun) [inline chain/byteexec/byteexec_test.go:25]: Add code as a comment here

**Review by taobun — CHANGES_REQUESTED** [review]

**Review by sorawit — APPROVED** [review]
Looks good

**Review by taobun — APPROVED** [review]

**Pull request merged**